src/work/WorkSequence.cpp

```diff
@@ -24,6 +25,13 @@ WorkSequence::onRun()
     if (mNextInSequence == mSequenceOfWork.end())
     {
         // Completed all the work
+        for (auto const& w : mSequenceOfWork)
```

Comment:
we have `allChildrenSuccessful()` for this

Reply:
No, you have the inheritence hierarchy wrong.

`BasicWork` is extended by `Work` that has a `std::list` called `Work::mChildren` and steps its children forwards in interleaved round-robin, removing children as they finish in any order. `Work::allChidlrenSuccessful` is a member of `Work`.

`BasicWork` is also, _separately_ extended by `WorkSequence` that stores its sequence in a `std::vector` called `WorkSequence::mSequenceOfWork` and steps its children forwards by running each to completion, one at a time, before advancing to the next. It has no member `allChildrenSuccessful`.

Reply:
We have some free functions that are used under the hood in `allChidlrenSuccessful`, see `WorkUtils::allSuccessful`. For the use case here, `WorkUtils::getWorkStatus` seems the most suitable (it returns success or failure for the list of work).

Reply:
@marta ok .. these take `std::list`, not `std::vector`. templatizing those helpers to make them container-agnostic seems like overkill; in this case I will just construct a temporary list at the call site from the vector.

Reply:
@marta ok .. these take std::list, not std::vector. templatizing those helpers to make them container-agnostic seems like overkill; in this case I will just construct a temporary list at the call site from the vector.